network.go

```diff
@@ -322,6 +322,32 @@ func getInterface(netHandle *netlink.Handle, link netlink.Link) (*pb.Interface,
 	return &ifc, nil
 }
 
+func (s *sandbox) listInterfaces(netHandle *netlink.Handle) (*pb.Interfaces, error) {
+	if netHandle == nil {
+		netHandle, err := netlink.NewHandle(unix.NETLINK_ROUTE)
+		if err != nil {
+			return nil, err
+		}
+		defer netHandle.Delete()
+	}
+
+	links, err := netHandle.LinkList()
+	if err != nil {
+		return nil, err
+	}
+
+	var interfaces pb.Interfaces
+	for _, link := range links {
+		ifc, err := getInterface(netHandle, link)
+		if err != nil {
+			agentLog.WithError(err).Error("listInterfaces() failed")
+			return &interfaces, err
+		}
+		interfaces.Interfaces = append(interfaces.Interfaces, ifc)
```

Comment:
I don't really like the syntax, but this is because of the grpc protocol...
Any way we could define a list of interfaces through grpc, so that it ends up like:
```go
type Interfaces []*Interface
```
instead of
```go
type Interfaces struct {
	Interfaces []*Interface `protobuf:"bytes,1,rep,name=Interfaces" json:"Interfaces,omitempty"`
}
```

Reply:
Yes. It is from grpc. Struct is needed to serialize . I just want to align with route.
```
 message Routes {
 	repeated Route Routes = 1;
 }
```
If grpc allows return a slice instead of a struct, I can make the change.